app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java

```diff
@@ -29,50 +28,50 @@
      */
     private DatabaseHelper(Context context) {
         super(context, DATABASE_NAME, null, 1);
-        DatabaseHelper.context = context;
+        this.context = context;
     }
 
     /**
      * Initializes values for database creation
      * @param context Interface to global information about an application environment.
      */
-    public static void setupDatabase(Context context) {
-        if (db == null) {
-            DatabaseHelper.context = context;
+    public static boolean setupDatabase(Context context) {
+        if (databaseHelper == null) {
             databasePath = context.getFilesDir().getPath() + "/../databases/";
+            databaseHelper = new DatabaseHelper(context);
+            return true;
+        } else {
+            return false;
         }
     }
 
     /**
      * @return A DatabaseHelper object
      */
-    public static DatabaseHelper getInstance() {
-        if (context == null) {
-            return null;
-        }
-        if (db == null) {
-            return databaseHelper = new DatabaseHelper(context);
+    public static DatabaseHelper getInstance() throws MCGADatabaseException {
+        if (databaseHelper == null) {
+            throw new MCGADatabaseException("Please setup the Database first.");
         } else {
             return databaseHelper;
         }
 
     }
 
     /**
-     *
-     * @return The database
+     * @return - an instance of a readable Database <b>DO NOT FORGET to call {@link
+     * SQLiteDatabase#close()}</b> after you are done!
      */
-    public static SQLiteDatabase getDb() {
-        return db;
+    public SQLiteDatabase getDb() {
+        return databaseHelper.getReadableDatabase();
     }
 
     /**
      * Creates a empty database on the system and rewrites it with our own database.
      */
-    public void createDatabase() throws IOException {
+    public void createDatabase(boolean force) throws IOException {
```

Comment:
Unneeded parameter, this is a code smell....

https://martinfowler.com/bliki/FlagArgument.html

Also, this method is only referenced once, so I do not see why it would need to have a flag.